src/sentry/static/sentry/app/views/settings/projectAlerts/create.tsx

```diff
@@ -39,14 +44,26 @@ class Create extends React.Component<Props, State> {
   };
 
   componentDidMount() {
-    const {organization, project} = this.props;
+    const {organization, project, hasCreateFromDiscover, location} = this.props;
 
     trackAnalyticsEvent({
       eventKey: 'new_alert_rule.viewed',
       eventName: 'New Alert Rule: Viewed',
       organization_id: parseInt(organization.id, 10),
       project_id: parseInt(project.id, 10),
     });
+
+    if (hasCreateFromDiscover && location.query.createFromDiscover) {
+      const eventView = EventView.fromLocation(location);
+      if (this.state.alertType !== 'metric') {
+        // eslint-disable-next-line react/no-did-mount-set-state
+        this.setState({alertType: 'metric', eventView});
```

Comment:
We should disable this rule in our eslint config (and for didUpdate too): https://github.com/getsentry/eslint-config-sentry